ksp_plugin.py

```diff
@@ -144,77 +155,92 @@ def read_file_function(self, filepath):
-        should_play_sound = settings.get('ksp_play_sound', False)
-
-        error_msg = None
-        error_lineno = None
-        error_filename = filepath # path to main script
-
-        sound_utility = CompilerSounds()
-
-        try:
-            sublime.status_message('Compiling...')
-
-            self.compiler = ksp_compiler.KSPCompiler(code, self.base_path, compact, compactVars,
-                                                     read_file_func=self.read_file_function,
-                                                     extra_syntax_checks=check,
-                                                     optimize=optimize and check,
-                                                     add_compiled_date_comment=add_compiled_date_comment)
-            if self.compiler.compile(callback=self.compile_on_progress):
-                last_compiler = self.compiler
-                code = self.compiler.compiled_code
-                code = code.replace('\r', '')
-                if self.compiler.output_file:
-                    if not os.path.isabs(self.compiler.output_file):
-                        self.compiler.output_file = os.path.join(self.base_path, self.compiler.output_file)
-                    codecs.open(self.compiler.output_file, 'w', 'latin-1').write(code)
-                    sublime.status_message("Successfully compiled (compiled code saved to %s)." % self.compiler.output_file)
+        for view in self.open_views:
+            self.current_view = view
+            code = view.substr(sublime.Region(0, view.size()))
+            filepath = view.file_name()
+            if filepath:
+                self.base_path = os.path.dirname(filepath)
+            else:
+                self.base_path = None
+
+            settings = sublime.load_settings("KSP.sublime-settings")
+
+            compact = settings.get('ksp_compact_output', False)
+            compactVars = settings.get('ksp_compact_variables', False)
+            check = settings.get('ksp_extra_checks', True)
+            optimize = settings.get('ksp_optimize_code', False)
+            add_compiled_date_comment = settings.get('ksp_add_compiled_date', True)
+            should_play_sound = settings.get('ksp_play_sound', False)
+
+            error_msg = None
+            error_lineno = None
+            error_filename = filepath # path to main script
+
+            sound_utility = CompilerSounds()
+           
+            pragma_compiled_source_re = re.compile(r'\{\s*\#pragma\s+save_compiled_source\s+(.*)\}')
+            if self.compile_all_open and not pragma_compiled_source_re.search(code):
+                if filepath == None:
+                    filepath = 'main script'
+                sublime.error_message('Attempted to compile multiple scripts, but no output path specified!\n\n' + '<'+filepath+'>')
+                sublime.status_message('Error no output path specified - compilation aborted!')
```

Comment:
Error: No output path was specified for script '%s' - compilation aborted!

Then show script's path instead of %s.